cli/cli/commands/portal/start/start.go

```
@@ -0,0 +1,53 @@
+package start
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/command_framework/lowlevel"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/command_framework/lowlevel/args"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/command_framework/lowlevel/flags"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/command_str_consts"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/portal_manager"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+
+var PortalStartCmd = &lowlevel.LowlevelKurtosisCommand{
+	CommandStr:               command_str_consts.PortalStartCmdStr,
+	ShortDescription:         "Starts Kurtosis Portal",
+	LongDescription:          "Starts Kurtosis Portal in the background. The portal can then be stopped or restarted using the corresponding commands",
+	Flags:                    []*flags.FlagConfig{},
+	Args:                     []*args.ArgConfig{},
+	PreValidationAndRunFunc:  nil,
+	RunFunc:                  run,
+	PostValidationAndRunFunc: nil,
+}
+
+func run(ctx context.Context, _ *flags.ParsedFlags, args *args.ParsedArgs) error {
+	logrus.Infof("Starting Kurtosis Portal")
+	portalManager := portal_manager.NewPortalManager()
+
+	// Checking if new version is available and downloading it
+	if _, err := portal_manager.DownloadLatestKurtosisPortalBinary(ctx); err != nil {
+		return stacktrace.Propagate(err, "Unable to download Kurtosis Portal binary")
```

do we always want to run portal with an active internet connection ; the server can be intranet

GitHub can be down. Can keep this for now if it makes life easier with versioning and such. 

Curious, will new portal work with any version of Kurtosis that has ever supported portal?

---

Yes I updated DownloadLatestKurtosisPortalBinary to not fail hard if a aversion is currently installed 👍 
And to your second point, I think yes, the portal API should be highly backward compatible, as mentioned in the docs. That might not be true at the beginning, but hopefully we can quickly trend toward something fairly stable. 
The responsibility of the portal is fairly narrow and well defined, right now, so hoping the API doesn't grow crazy

We will see, worst case we can add a flag to request an upgrade. Or we can even encode somewhere in our code the compatibility between Kurtosis and the Portal.